sys/vm/vm_mmap.c

```diff
@@ -772,14 +776,14 @@ kern_mmap(struct thread *td, const struct mmap_req *mrp)
 		 * with maxprot later.
 		 */
 		cap_rights_init_one(&rights, CAP_MMAP);
-		if (max_prot & PROT_READ)
+		if (cap_prot & PROT_READ)
 			cap_rights_set_one(&rights, CAP_MMAP_R);
 		if ((flags & MAP_SHARED) != 0) {
-			if (max_prot & PROT_WRITE)
+			if (cap_prot & PROT_WRITE)
 				cap_rights_set_one(&rights, CAP_MMAP_W);
+			if (cap_prot & PROT_EXEC)
```

Comment:
You shouldn't need to move this any more

Reply:
right, I testted it still works without the move